crates/libs/bindgen/src/winmd/verify.rs

```diff
@@ -2,6 +2,18 @@ use super::*;
 use metadata::RowReader;
 
 pub fn verify(reader: &metadata::Reader, filter: &metadata::Filter) -> crate::Result<()> {
+    let unused: Vec<&str> = filter.unused(reader).collect();
+
+    if !unused.is_empty() {
+        let mut message = "unused filters".to_string();
+
+        for unused in unused {
+            message.push_str(&format!("\n  {unused}"));
+        }
+
+        return Err(crate::Error::new(&message));
```

Comment:
Surfacing these are errors versus warnings may make crate third party crate maintenance harder than it needs to be. I can see a scenario where folks will want to add a preemptive filter for, say, X.Y.Z.Preview namespaces when they appear/don't appear during a dev cycle. Or simply don't care if they over-specify the filter list.

No change needed right now though, we can take the "lets see what happens" approach.

Reply:
Personally, I do prefer the error because I'd like the filter list to reflect what goes into the generated bindings (no more, no less). But I accept that other people may have other preferences.